File: 350.py
from collections import Counter

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        count1, count2 = Counter(nums1), Counter(nums2)
        res = []
        for key, value in count1.items():
            min_count = min(value, count2[key])
            for i in range(min_count):
                res.append(key)
        return res 

File: 350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> count1;
        vector<int> res;

        for(int i = 0; i < nums1.size(); i++) {
            count1[nums1[i]]++;
        }
        for(int i = 0; i < nums2.size(); i++) {
            if(count1[nums2[i]] > 0) {
                count1[nums2[i]]--;
                res.push_back(nums2[i]);
            }
        }

        return res;
    }
};